infrastructure/smart-contracts/contracts/upgrades/referee/Referee9.sol

```diff
@@ -424,16 +425,61 @@ contract Referee9 is Initializable, AccessControlEnumerableUpgradeable {
         bytes32 comboHash = keccak256(abi.encodePacked(_assertionId, rollupAddress));
         require(!rollupAssertionTracker[comboHash], "9");
         rollupAssertionTracker[comboHash] = true;
+        
+        // Connect to the rollup contract
+        IRollupCore rollup = IRollupCore(rollupAddress);
+
+        // get the node information from the rollup.
+        Node memory currentNode = rollup.getNode(_assertionId);
+
+
+
+        // If the gap is more than 1 assertion, we need to handle as a batch challenge
+        if(_assertionId - _predecessorAssertionId > 1){
+
+            // Initialize the array to store the assertionIds
+            uint64 [] memory assertionIds = new uint64[](_assertionId - _predecessorAssertionId);
+
+            // Loop through the assertions and check if they have been submitted
+            for(uint64 i = _predecessorAssertionId + 1; i < _assertionId; i++){
+
+                // create the comboHash for the assertionId and rollupAddress
+                bytes32 comboHashBatch = keccak256(abi.encodePacked(i, rollupAddress));
+
+                // check the assertionId and rollupAddress combo haven't been submitted yet
+                require(!rollupAssertionTracker[comboHashBatch], "15");
```

Comment:
This should be error `"9"`
It seems we will be checking that twice since its already done a couple lines above 

```
bytes32 comboHash = keccak256(abi.encodePacked(_assertionId, rollupAddress));
require(!rollupAssertionTracker[comboHash], "9");
```

Reply:
Error number corrected. 
We are not checking twice as the loop does not include the last assertion id.